composer/loggers/mlflow_logger.py

```diff
@@ -39,67 +44,97 @@ def __init__(
         run_name: Optional[str] = None,
         tracking_uri: Optional[Union[str, pathlib.Path]] = None,
         rank_zero_only: bool = True,
+        flush_interval: int = 10,
     ) -> None:
         try:
             import mlflow
+            from mlflow import MlflowClient
+            from mlflow.utils.autologging_utils import MlflowAutologgingQueueingClient
         except ImportError as e:
             raise MissingConditionalImportError(extra_deps_group='mlflow',
                                                 conda_package='mlflow',
                                                 conda_channel='conda-forge') from e
-        del mlflow
         self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
 
         self.run_name = run_name
         self.experiment_name = experiment_name
         self._rank_zero_only = rank_zero_only
-        self.tracking_uri = tracking_uri
+        self.tracking_uri = str(tracking_uri or mlflow.get_tracking_uri())
+        self._last_flush_time = time.time()
+        self._flush_interval = flush_interval
+        # Set up MLflow state
+        self._run_id = None
+        if self.experiment_name is None:
+            self.experiment_name = os.getenv(mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.name,
+                                             DEFAULT_MLFLOW_EXPERIMENT_NAME)
+        self._mlflow_client = MlflowClient(self.tracking_uri)
+        # Create an instance of MlflowAutologgingQueueingClient - an optimized version
+        # of MlflowClient - that automatically batches metrics together and supports
+        # asynchronous logging for improved performance
+        self._optimized_mlflow_client = MlflowAutologgingQueueingClient(self.tracking_uri)
+        # set experiment. we use MlflowClient for experiment retrieval and creation
+        # because MlflowAutologgingQueueingClient doesn't support it
+        env_exp_id = os.getenv(mlflow.environment_variables.MLFLOW_EXPERIMENT_ID.name, None)
+        if env_exp_id is not None:
+            self._experiment_id = env_exp_id
+        elif exp := self._mlflow_client.get_experiment_by_name(name=self.experiment_name):
```

Comment:
```suggestion
        elif exp := self._mlflow_client.get_experiment_by_name(name=self.experiment_name):
```
nit: can we not use walrus? 

we officially only support >=3.8, but it would be nice to be backwards compatible (up until now we are!) for community users who are using lower python versions for compatibility with other libraries

Reply:
Great point! I've removed the walrus! I am no longer the walrus. 🦭